ixmp/core.py

```diff
@@ -164,6 +164,51 @@ def scenario_list(self, default=True, model=None, scen=None):
         return pd.DataFrame(self._backend.get_scenarios(default, model, scen),
                             columns=FIELDS['get_scenarios'])
 
+    def export_timeseries_data(self, path, default=True, model=None,
+                               scenario=None, variables=None):
+        """Export timeseries data to CSV file across multiple scenarios.
+
+        Refer :meth:`.add_timeseries` of :class:`Timeseries` to get more
+        information about adding timeseries data to scenario.
+
+        Parameters
+        ----------
+        path : os.PathLike
+            File name to export data to; must have the suffix '.csv'.
+
+            Result file will contain the following columns:
+
+            - model
+            - scenario
+            - version
+            - variable
+            - unit
+            - region
+            - meta
+            - time
+            - year
+            - value
+        default : bool, optional
+            :obj:`True` to include only TimeSeries versions marked as default.
+        model : str or None, optional
+            Model name to filter results.
+        scenario : str or None, optional
+            Scenario name to filter results.
+        variables : list, optional
+            List of timeseries variables (names) to export.
+
+        Returns
+        -------
+        None
+        """
+        filters = {
+            'model': model,
+            'scenario': scenario,
+            'variable': as_str_list(variables) or [],
+        }
+
+        self._backend.write_file(path, ItemType.TS, filters, default=default)
```

Comment:
@khaeru is there any particular reason not putting `default` (or `is_default` if better) to filters?

Reply:
Elsewhere the code, `filters` are always a `dict` where the keys are the dimensions and the values are `list of str` with allowable values. Only the values appearing in the filters are returned.

You could think of `default` as an attribute that has a True or False value. But elsewhere in the code (e.g. Platform.scenarios_list), a `default=False` argument does *not* mean "_only_ scenarios that have default=False"; instead, it means "all scenarios, regardless of whether default=False or default=True".

I personally think that is confusing, but it's extra confusing to do it two different ways in two places. So that's why, for now, I would keep these separate.